internal/databases/postgres/backup_push_handler.go

```diff
@@ -682,6 +684,55 @@ func (bh *BackupHandler) checkDataChecksums() error {
 	return nil
 }
 
+// CheckArchiveCommand verifies the archive_mode and archive_command settings.
+// It logs warnings if WAL archiving is not properly configured.
+func (bh *BackupHandler) CheckArchiveCommand() error {
+	// Check if the server is in recovery mode (standby)
+	standby, err := bh.Workers.QueryRunner.IsStandby()
+	if err != nil {
+		tracelog.ErrorLogger.Printf("CheckArchiveCommand: failed to determine standby mode: %v", err)
+		return err
+	}
+
+	if standby {
+		// If the server is in standby mode, no further checks are needed
+		tracelog.DebugLogger.Println("Server is in standby mode. Skipping archive settings checks.")
+		return nil
+	}
+
+	// Retrieve the current archive_mode setting
+	archiveMode, err := bh.Workers.QueryRunner.GetArchiveMode()
+	if err != nil {
+		tracelog.ErrorLogger.Printf("CheckArchiveCommand: failed to get archive_mode: %v", err)
+		return err
+	}
+
+	// Check if archive_mode is enabled
+	if archiveMode != "on" && archiveMode != "always" {
+		tracelog.WarningLogger.Println(
+			"archive_mode is not enabled. This may cause inconsistent backups. " +
+				"Please consider configuring WAL archiving.")
+	} else {
+		// Retrieve the current archive_command setting
+		archiveCommand, err := bh.Workers.QueryRunner.GetArchiveCommand()
+		if err != nil {
+			tracelog.ErrorLogger.Printf("CheckArchiveCommand: failed to get archive_command: %v", err)
+			return err
+		}
+
+		// Check if archive_command is properly configured
+		if len(archiveCommand) == 0 || archiveCommand == "(disabled)" {
+			tracelog.WarningLogger.Println(
+				"archive_command is not configured. This may cause inconsistent backups. " +
+					"Please consider configuring WAL archiving.")
+		} else {
+			tracelog.DebugLogger.Println("WAL archiving settings are configured.")
+		}
+	}
+
+	return nil
+}
```

Comment:
It seem that after this change we will not show warning for greenplum:
https://github.com/wal-g/wal-g/blob/v3.0.5/internal/databases/greenplum/backup_push_handler.go#L484

Reply:
> It seem that after this change we will not show warning for greenplum: https://github.com/wal-g/wal-g/blob/v3.0.5/internal/databases/greenplum/backup_push_handler.go#L484

@ostinru Hi 
I've [added](https://github.com/wal-g/wal-g/pull/1882/commits/a5b0ec22a112dfa243a369e1bf6d191d4d81050d) 'Archive mode' check into Greenplum.

In this commit, I decided to define a separate `func CheckArchiveCommand(conn *pgx.Conn) error` specifically for the Greenplum code. Currently, it references methods from the postgres package. This separation is intended to allow future modifications or overrides of these methods or logic for Greenplum without impacting the postgres implementation.

Reply:
@ostinru Hi could you review my last changes (comment above)?